### PR TITLE
[YD-2292] Ability to set the session URL by introducing it in a textview

### PR DESCRIPTION
**Pull request by mariasenosiain**
Related to [YD-2292](https://lampkicking.atlassian.net/browse/YD-2292)

B&T

- [x] Complete session flow:
1. Generate a session URL with the session tool.
2. Run `webapp` and shake the phone.
3. BottomSheet appears, paste the session URL in the text field.
4. Tap on `START SESSION`.
5. Session is started and you can complete the flow.
6. Finish session dialog appears when final URL is loaded (yoti.com web).

### CODE REVIEW

File: webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/MainActivity.kt [hunk 1]
```diff
@@ -246,21 +263,34 @@ class MainActivity : AppCompatActivity() {
...
-                return true
             }
-            request?.url?.toString()?.let { Log.d(TAG, it) }
-
-            return super.shouldOverrideUrlLoading(view, request)
         }
     }
+
+    private fun showOptionsDialog() {
+        if (sessionBottomSheet != null) return
```
Comment (damienlo) [inline webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/MainActivity.kt:279]: should we consider checking `sharedialog.isShowing` before return?

File: webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/AccelerometerListener.kt [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    }
+
+    override fun onSensorChanged(event: SensorEvent?) {
+        event?.run {
+            val x = values[0]
+            val y = values[1]
+            val z = values[2]
+            lastShake = currentShake
+            currentShake = sqrt((x * x + y * y + z * z).toDouble()).toFloat()
+            val delta: Float = currentShake - lastShake
```
Comment (damienlo) [inline webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/AccelerometerListener.kt:49]: if we make it a double. it avoids a double conversion

File: webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/MainActivity.kt (hunk 1 above)
Comment (mariasenosiain) [reply webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/MainActivity.kt:279]: We're setting it to null when it is dismissed, so I think that verification is not needed :)

File: webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/AccelerometerListener.kt (hunk 2 above)
Comment (mariasenosiain) [reply webapp/src/main/java/com/yoti/mobile/android/sdk/yotidocscan/websample/AccelerometerListener.kt]: Changed, `sqrf` function is already returning Float, so conversion to double was not needed either 😅

**Comment by ericluyoti** [comment]
B&T passed.

**Pull request merged**